src/msw/notebook.cpp

```diff
@@ -729,22 +730,244 @@ bool wxNotebook::InsertPage(size_t nPage,
     return true;
 }
 
+#ifdef WXDEBUG_NOTEBOOK_HITTEST
+// This is a debug function to visualize the hitbox when debugging the HitTest() function.
+static void ShowHitbox(wxWindow *w, wxRect r, bool text, bool refresh)
+{
+    wxBrush green(wxColor(0, 200, 0));
+    wxBrush blue(wxColor(0, 0, 200));
+
+    wxScreenDC dc;
+
+    if (refresh)
+    {
+        w->Refresh();
+        w->Update();
+    }
+
+    r.x += (r.width / 2)+2;
+    w->ClientToScreen(&r.x, &r.y);
+    if (text)
+        dc.SetBrush(green);
+    else
+        dc.SetBrush(blue);
+
+    dc.DrawRectangle(r);
+}
+#endif  // WXDEBUG_NOTEBOOK_HITTEST
+
+int wxNotebook::MSWHitTestLeftRight(const wxPoint &pt, long *flags) const
+{
+    // In Windows there is a bug so that the flags are not properly reported
+    // for left/right orientation, so we have to do it manually.
+    // It seems that the tabs are evaluated the same as if they were top/bottom
+    // oriented, because if multiple tabs are stacked the flags are quite different
+    // for each tab on the x axis.
+    wxDirection tabDir = GetTabOrientation();
+    //if (!(tabDir == wxLEFT || tabDir == wxRIGHT))
+    //    return false;
+
+    long dummy;
+    if (!flags)
+        flags = &dummy;
+
+    *flags = 0;
+
+    wxRect r = GetRect();
+    if (!r.Contains(pt))
+    {
+        *flags = wxBK_HITTEST_NOWHERE;
+        return -1;
+    }
+
+    size_t pages = GetPageCount();
+    if (!pages)
+    {
+        *flags |= wxBK_HITTEST_NOWHERE | wxBK_HITTEST_ONPAGE;
+        return -1;
+    }
+
+    if (GetPageSize().Contains(pt))
+    {
+        *flags = wxBK_HITTEST_ONPAGE;
+        return -1;
+    }
+
+    for (size_t i = 0; i < pages; i++)
+    {
+        r = GetTabRect(i);
+        if (!r.Contains(pt))
+            continue;
+
+        int item = i;
+
+        // If the user doesn't care about the flags we are done here.
+        if (flags == &dummy)
+            return item;
+
+        // Just a magic number between the icon and the start of the text
+        // I checked with different resolutions and it seems to be about right.
+        // My assumption was that the icon and the label are both centered in the tab
+        // and this assumption seems to be mostly right with this adjustemnt offest
+        // being required.s
+        int gap = 8;
+
+        TCHAR buffer[256];
+        TCITEM ti;
+        ti.mask = TCIF_IMAGE | TCIF_TEXT;
+        ti.cchTextMax = sizeof(buffer) - 1;
```

Comment:
I just wanted to point out one important change among several I made in this code: `sizeof(buffer)` is wrong here, it's the size in bytes, i.e. double of the actual buffer size in `TCHAR`s and so could theoretically result in a buffer overflow here if we had a label longer than 128 characters.